### PR TITLE
show error & disable dynamic scan if device not available

### PR DESCRIPTION
**Pull request by future-pirate-king**
## Summary by CodeRabbit

- **New Features**
	- Introduced a component for managing device preferences in projects, allowing selection of device type and OS version.
	- Added a new `ProjectPreferences::Provider` component for dynamic scan scheduling and API settings adjustments.
- **Bug Fixes**
	- Enhanced error handling in device selection to manage unavailable device types or versions.
- **Documentation**
	- Added translations for messages related to unavailable device preferences in English and Japanese.
- **Refactor**
	- Refactored `ProjectPreferencesComponent` to improve structure and logic related to device preferences.
- **Style**
	- Updated modal design to accommodate new settings and preferences management.
- **Tests**
	- Added integration tests for dynamic scan functionality and device preference management.

### CODE REVIEW

File: app/components/project-preferences/provider/index.ts [hunk 1]
```diff
@@ -0,0 +1,214 @@
+import Component from '@glimmer/component';
+import { inject as service } from '@ember/service';
+import ENUMS from 'irene/enums';
+import ENV from 'irene/config/environment';
```
Comment (Yibaebi) [inline app/components/project-preferences/provider/index.ts:4]: Group related imports together.

File: app/components/project-preferences/provider/index.ts [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+  @tracked selectedDeviceType?: DeviceType;
+
+  @tracked deviceTypes = ENUMS.DEVICE_TYPE.CHOICES;
+  @tracked devicePreference?: DevicePreferenceModel;
+
+  @tracked
+  devices: DS.AdapterPopulatedRecordArray<ProjectAvailableDeviceModel> | null =
+    null;
+
+  get tAnyVersion() {
```
Comment (Yibaebi) [inline app/components/project-preferences/provider/index.ts:59]: Position of getter should be after constructor.

File: app/components/project-preferences/provider/index.ts (hunk 1 above)
Comment (future-pirate-king) [reply app/components/project-preferences/provider/index.ts]: done

File: app/components/project-preferences/provider/index.ts (hunk 2 above)
Comment (future-pirate-king) [reply app/components/project-preferences/provider/index.ts]: done

**Review by Yibaebi — APPROVED** [review]

**Pull request merged**